File: benchmark_SRVR.js
#!/usr/bin/env node

function rnd (n) { return Math.floor(n* Math.random()) }

var words= "Con diez cañones por banda viento en\
 popa a toda vela no corta el mar si no vuela un\
 velero bergantín bajel pirata llamado por su bravura\
 el temido en todo el mar conocido del uno al otro confín".toLowerCase().split(" ");

var words_length = words.length;

function poema (length, r, curr, prev, l) {
  r= [];
  l= 0;
  while (l < length) {
    do curr= words[rnd(words_length)]; while (curr === prev);
    r.push(prev= curr);
    l+= prev.length+ 1;
  }
  return r.join(" ");
}

var port= 8081;
require('http').createServer(callback).listen(port);
var res= new Buffer("HOLA");
var headers= {Server: "NODE", "Content-Length": res.length, "Connection": "close"};
function callback (request, response) {
  response.writeHead(200, headers);
  response.end(res);
}

console.log("NODE Server running on port "+ port);

File: benchmark_SRVR.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'eventmachine'

class RequestHandler < EM::P::HeaderAndContentProtocol
  
  $words='Con diez cañones por banda viento en popa a toda vela
  no corta el mar si no vuela un velero bergantín bajel
  pirata llamado por su bravura el temido en todo el mar
  conocido del uno al otro confín'.downcase.split

  $words_length = $words.length

  def poema(length)
    r=[]
    prev=''
    curr=''
    l = 0

    while (l < length) do
       begin
         curr=$words[rand($words_length)]  
       end until (curr != prev)

        l = l + curr.length + 1
        r << (prev=curr)
    end

    return r.join(' ')
  end
  
  def receive_request headers, content
    unPoema= poema(4096)
    send_data("HTTP/1.1 200 OK\nContent-Length: #{unPoema.length}\nContent-Type: text/plain\nServer: RUBY\n\n#{unPoema}")
    close_connection_after_writing
  end
end

EM.run{
  port = 8080
  EM.start_server 'localhost', port, RequestHandler
  puts "RUBY Server running on port #{port}"
}

